dev/tools/controllerbuilder/pkg/updatetypes/main.go

```diff
@@ -0,0 +1,235 @@
+func findDependentMsgs(field protoreflect.FieldDescriptor, ignoredProtoFields []string) (map[string]protoreflect.MessageDescriptor, error) {
+	deps := make(map[string]protoreflect.MessageDescriptor)
+	analyzeDependencies(field, deps, ignoredProtoFields)
+
+	// fmt.Println("\nDependencies:")
+	// for _, dep := range deps {
+	// 	fmt.Printf("%s\n", dep.FullName())
+	// }
+	return deps, nil
+}
+
+// analyzeDependencies recursively explores the dependent proto messages of the given field.
+func analyzeDependencies(field protoreflect.FieldDescriptor, deps map[string]protoreflect.MessageDescriptor, ignoreFields []string) {
+	if shouldIgnore(string(field.FullName()), ignoreFields) {
+		return
+	}
+
+	if field.IsMap() {
+		/* an example Map field
+		message ParentMessage {
+			map<KeyMessage, ValueMessage> some_map_field = 1;
+		} */
+		mapEntry := field.Message()
+		if keyField := mapEntry.Fields().ByName("key"); keyField != nil {
+			analyzeDependencies(keyField, deps, ignoreFields)
+		}
+		if valueField := mapEntry.Fields().ByName("value"); valueField != nil {
+			analyzeDependencies(valueField, deps, ignoreFields)
+		}
+	} else {
+		switch field.Kind() {
+		case protoreflect.MessageKind:
+			msg := field.Message()
+			fqn := string(msg.FullName())
+			if _, ok := deps[fqn]; !ok {
+				deps[fqn] = msg
+				for i := 0; i < msg.Fields().Len(); i++ {
+					field := msg.Fields().Get(i)
+					analyzeDependencies(field, deps, ignoreFields)
+				}
+			}
+		case protoreflect.EnumKind:
+			// deps[string(field.Enum().FullName())] = true  // Skip enum because enum is mapped to Go string in code generation
+		}
+	}
+}
+
+func shouldIgnore(fieldName string, ignoreFields []string) bool {
+	for _, ignoreField := range ignoreFields {
+		if fieldName == ignoreField {
+			return true
+		}
+	}
+	return false
+}
+
+func removeNotMappedToGoStruct(msgs map[string]protoreflect.MessageDescriptor) {
+	for _, msgName := range protoMsgsNotMappedToGoStruct {
+		delete(msgs, msgName)
+	}
+}
+
+// removeAlreadyGenerated removes proto messages that have already been generated (including manually edited)
+func removeAlreadyGenerated(goPackagePath, outputAPIDirectory string, targets map[string]protoreflect.MessageDescriptor) error {
+	packages, err := gocode.LoadPackageTree(goPackagePath, outputAPIDirectory)
+	if err != nil {
+		return err
+	}
+	for _, p := range packages {
+		for _, s := range p.Structs {
+			if annotation := s.GetAnnotation("+kcc:proto"); annotation != "" {
+				if _, ok := targets[annotation]; ok {
+					delete(targets, annotation)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func writeToLocalFile(field protoreflect.FieldDescriptor, parentMsg protoreflect.MessageDescriptor, msgs map[string]protoreflect.MessageDescriptor) error {
+	file, err := os.Create("output.txt")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	fmt.Fprintf(file, "[The generated new field]\n\n")
```

Comment:
Small idea: I suggest being a little more verbose here, and possibly even printing a "link" to the file & line number if you know it.  (Many editors recognize [path]:[line-number], I believe)

Big idea: You could also try feeding those more verbose instructions into AI :-)  You could feed it in manually a few times to mess around with it and the prompt, and if that works ... maybe offer a mode where we try to apply the change!

Reply:
Sounds good! I left a TODO in the code. Manually inserting the generated code back to the files was never the design intention. I am finding a way do this automatically.